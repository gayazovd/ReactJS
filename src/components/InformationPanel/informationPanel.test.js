import React from 'react';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import InformationPanelView from './view';

configure({ adapter: new Adapter() });

describe('function onClick call', () => {
  const onHandleClick = jest.fn();
  const props = {
    onHandleClick,
    tabs: [{ id: 1 }, { id: 2 }],
    activeTab: '1'
  };
  it('snapshot', () => {
    const informationPanelView = shallow(<InformationPanelView {...props} />);
    expect(informationPanelView).toMatchSnapshot();
  });

  it('function called', () => {
    const informationPanelView = shallow(<InformationPanelView {...props} />)
      .find('.activeTab')
      .at(1);
    informationPanelView.simulate('click');
    expect(onHandleClick).toHaveBeenCalledTimes(1);
  });
});
