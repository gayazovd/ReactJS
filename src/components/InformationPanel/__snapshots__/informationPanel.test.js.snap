// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`function onClick call snapshot 1`] = `
<Fragment>
  <div>
     movies found
  </div>
  <div
    className="sort-oprions"
  >
    <div>
      Sort by
    </div>
    <div
      className="activeTab"
      key="1"
      onClick={[Function]}
    />
    <div
      className="activeTab"
      key="2"
      onClick={[Function]}
    />
  </div>
</Fragment>
`;
